lib/tapioca/dsl/compilers/active_record_relations.rb

```diff
@@ -699,10 +700,11 @@ def create_common_methods
                   *(create_kw_opt_param("order", type: "Symbol", default: ":asc") if order),
                   create_block_param("block", type: "T.nilable(T.proc.params(object: #{constant_name}).void)"),
                 ],
-                return_type: "T.nilable(T::Enumerator[#{constant_name}])",
+                return_type: weak_find_each?(constant_name) ? as_nilable_type(return_type) : return_type,
```

Comment:
It seems this is changing the default behaviour - can we flip it to be the other way?